packages/compass-global-writes/src/components/shard-zones-table.tsx

```diff
@@ -8,47 +8,189 @@ import {
   Row,
   Cell,
   css,
+  useLeafyGreenTable,
+  type LGColumnDef,
+  type LeafyGreenTableRow,
+  flexRender,
+  type HeaderGroup,
+  SearchInput,
+  type LGTableDataType,
+  getFilteredRowModel,
+  type LgTableRowType,
 } from '@mongodb-js/compass-components';
 import type { ShardZoneData } from '../store/reducer';
 
 const containerStyles = css({
-  maxWidth: '700px',
   height: '400px',
 });
 
+interface ShardZoneRow {
+  locationName: string;
+  zone: string;
+}
+
+interface ShardZoneExpandableRow extends ShardZoneRow {
+  subRows: ShardZoneRow[];
+}
+
+const columns: Array<LGColumnDef<ShardZoneRow>> = [
+  {
+    accessorKey: 'locationName',
+    header: 'Location Name',
+    enableSorting: true,
+  },
+  {
+    accessorKey: 'zone',
+    header: 'Zone',
+    enableSorting: true,
+    enableGlobalFilter: false,
+  },
+];
+
+const parseRow = ({
+  isoCode,
+  readableName,
+  zoneName,
+  zoneLocations,
+}: ShardZoneData): ShardZoneRow => ({
+  locationName: `${readableName} (${isoCode})`,
+  zone: `${zoneName} (${zoneLocations.join(', ')})`,
+});
+
+const parseData = (shardZones: ShardZoneData[]): ShardZoneExpandableRow[] => {
+  const grouppedZones = shardZones.reduce<
+    Record<ShardZoneData['typeOneIsoCode'], ShardZoneExpandableRow>
+  >((groups, next) => {
+    const { typeOneIsoCode, isoCode } = next;
+    if (!groups[typeOneIsoCode]) {
+      groups[typeOneIsoCode] = { ...parseRow(next), subRows: [] };
+    }
+    if (typeOneIsoCode === isoCode) {
+      Object.assign(groups[typeOneIsoCode], parseRow(next));
+    } else {
+      groups[typeOneIsoCode].subRows.push(parseRow(next));
+    }
+    return groups;
+  }, {});
+  return Object.values(grouppedZones);
+};
+
+const hasFilteredChildren = (
+  row: LgTableRowType<LGTableDataType<ShardZoneRow>>
```

Comment:
Is my understanding here correct that the LG table component is based on top of the `@tanstack/table-core` package and we're only using the latter because LG doesn't expose a type like it? Do you think it's clear enough what the different between e.g. `LgTableRowType` and `LeafygreenTableRow` is?

If that's the case, I'd assume that LG either doesn't re-export relevant types by accident (i.e. we could ask them for that in a ticket and leave a TODO comment) or intentionally (in which case we shouldn't be using them and considering them an implementation detail, right?).

We could get away with a more descriptive type here as an alternative, e.g.

```suggestion
  row: { subRows: { columnFilters: Record<string, unknown> }[] }
```

Reply:
So I was digging into this and they do reexport the tanstack table, the trouble is some of the exports (like the Row) get overridden by other exports (LG table defines it's own Row type). The LG team said they are doing some refactor on the table right now, so they'll look into this

Reply:
Alright – then I'd add a TODO comment with that reference, or go with something like the suggestion above here (I think that's perfectly fine typing for this situation)

Reply:
Added small ticket for us, linked to their refactor ticket: https://jira.mongodb.org/browse/COMPASS-8437 (waiting for confirmation from Shaneeza that I got the correct ticket)